src/main/java/gregicadditions/machines/multi/GAFuelRecipeLogic.java

```diff
@@ -0,0 +1,128 @@
+import java.util.function.Supplier;
+
+public class GAFuelRecipeLogic extends FuelRecipeLogic {
+    public final FuelRecipeMap recipeMap;
+    protected FuelRecipe previousRecipe;
+    protected final Supplier<IEnergyContainer> energyContainer;
+    protected final Supplier<IMultipleTankHandler> fluidTank;
+    public final long maxVoltage;
+    private int recipeDurationLeft;
+    private long recipeOutputVoltage;
+    private boolean workingEnabled = true;
+    private boolean wasActiveAndNeedsUpdate = false;
+
+    public GAFuelRecipeLogic(MetaTileEntity metaTileEntity, FuelRecipeMap recipeMap, Supplier<IEnergyContainer> energyContainer, Supplier<IMultipleTankHandler> fluidTank, long maxVoltage) {
+        super(metaTileEntity, recipeMap, energyContainer, fluidTank, maxVoltage);
+        this.recipeMap = recipeMap;
+        this.energyContainer = energyContainer;
+        this.fluidTank = fluidTank;
+        this.maxVoltage = maxVoltage;
+    }
+
+    @Override
+    protected int calculateRecipeDuration(FuelRecipe currentRecipe) {
+        if (metaTileEntity instanceof GAFueledMultiblockController) {
+            GAFueledMultiblockController gaController = (GAFueledMultiblockController) metaTileEntity;
+            if (gaController.hasMaintenanceHatch()) {
+                int numMaintenanceProblems = gaController.getNumProblems();
+                double maintenanceDurationMultiplier = 1.0 - (0.2 * numMaintenanceProblems);
+                int durationModified = (int) (currentRecipe.getDuration() * maintenanceDurationMultiplier);
+
+                gaController.calculateMaintenance(durationModified);
+                return durationModified;
+            }
+        }
+
+        return currentRecipe.getDuration();
+    }
+
+    @Override
+    public void update() {
+        if (!this.getMetaTileEntity().getWorld().isRemote) {
+            if (this.workingEnabled) {
+                if (this.recipeDurationLeft > 0 && (this.energyContainer.get().getEnergyCanBeInserted() >= this.recipeOutputVoltage || this.shouldVoidExcessiveEnergy())) {
+                    this.energyContainer.get().addEnergy(this.recipeOutputVoltage);
+                    if (--this.recipeDurationLeft == 0) {
+                        this.wasActiveAndNeedsUpdate = true;
+                    }
+                }
+
+                if (this.recipeDurationLeft == 0 && this.isReadyForRecipes()) {
+                    // do not run recipes if no problems fixed
+                    if (metaTileEntity instanceof GAFueledMultiblockController) {
+                        GAFueledMultiblockController gaController = (GAFueledMultiblockController) metaTileEntity;
+                        if (gaController.hasMaintenanceHatch()) {
+                            int numMaintenanceProblems = gaController.getNumProblems();
+                            if (numMaintenanceProblems != 6)
+                                this.tryAcquireNewRecipe();
+                        }
```

Comment:
Needs to call tryAcquireNewRecipe if the multiblock is an instance of a Fueled multiblock, but does not have maintenance.

```suggestion
                        } else {
                             this.tryAcquireNewRecipe();
                        }
```